src/templates/console.html

```diff
@@ -11,51 +11,56 @@
         window.languagePluginUrl = '{{ PYODIDE_BASE_URL }}';
     </script>
     <script src="{{ PYODIDE_BASE_URL }}pyodide.js"></script>
+    <style>
+      .terminal { --size: 1.5; }
+    </style>
   </head>
   <body>
     <script>
       languagePluginLoader.then(() => {
+        pyodide.runPython(`
+            from pyodide import console
+            import js
+
+
+            class Console(console.InteractiveConsole):
+                def runcode(self, code):
+                    self.redirect_stdstreams()
+                    js.term.pause()
+                    js.term.runPython("\\n".join(self.buffer))
+
+                def banner(self):
+                    return f"Welcome to the Pyodide terminal emulator 🐍\\n{super().banner()}"
+
+
+            # JQuery.terminal add trailing newline that is not easy to remove
+            # with the current API. Doing our best here.
+            _c = Console(stdout_callback=lambda s: js.term.echo(s.rstrip("\\n")),
+                         stderr_callback=lambda s: js.term.error(s.rstrip("\\n")),
+                         persistent_stream_redirection=False)
+        `)
+
+        let c = pyodide.pyimport('_c');
+
         function pushCode(line) {
-          handleResult(c.push(line))
+          handleResult(c.push(line));
         }
 
-        let term = $('body').terminal(
+        var term = $('body').terminal(
```

Comment:
```suggestion
        let term = $('body').terminal(
```
unless it was intentional change?

Reply:
No, probably forgotten when merging with #979.